src/main/java/org/example/gabartischolarsqbbe/service/AuthService.java

```diff
@@ -0,0 +1,50 @@
+package org.example.gabartischolarsqbbe.service;
+
+import org.example.gabartischolarsqbbe.entity.User;
+import org.example.gabartischolarsqbbe.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import java.util.Optional;
+
+@Service
+public class AuthService {
+
+    @Autowired
+    private UserRepository userRepository;
```

Comment:
A major problem here is that we communicating with the `UserRepository` direct, we should have a `UserService` class to handle the business logic and to be the only allowed class to use the `UserRepository`